routers/frontend/routes.go

```diff
@@ -58,35 +64,48 @@ func (r *frontendRouter) renderProfilePage(ctx *gin.Context, statusCode int, err
 	}
 }
 
-func (r *frontendRouter) getBasicUserInfo(ctx *gin.Context, jwt string) (basicUserInfo, error) {
-	user, err := r.userService.GetUserWithJWT(ctx, jwt)
+func (r *frontendRouter) getProfilePageData(ctx *gin.Context, jwt string) (profilePageData, error) {
+	var data profilePageData
+	var err error
+	data.User, err = r.userService.GetUserWithJWT(ctx, jwt)
 	if err != nil {
-		return basicUserInfo{}, err
+		return profilePageData{}, err
 	}
-	var team *entities.Team
-	var teammates []entities.User
-	if user.Team != primitive.NilObjectID {
-		team, err = r.teamService.GetTeamWithID(ctx, user.Team.Hex())
-		if err != nil {
-			return basicUserInfo{
-				User: user,
-			}, nil
-		}
-		teammates, err = r.userService.GetTeammatesForUserWithID(ctx, user.ID.Hex())
+
+	if data.User.AuthLevel >= authlevels.Organizer {
+		users, err := r.userService.GetUsers(ctx)
 		if err != nil {
-			return basicUserInfo{
-				User:      user,
-				Team:      team,
-				Teammates: []entities.User{},
-			}, nil
+			r.logger.Error("could not get users", zap.Error(err))
+			return data, nil
 		}
+		data.AdminData.Users = users
+	}
+
+	data.Team, data.Teammates, err = r.getUserTeamInfo(ctx, data.User)
+	if err != nil {
+		r.logger.Error("could not get information about user's team", zap.Error(err))
+		return data, nil
 	}
 
-	return basicUserInfo{
-		User:      user,
-		Team:      team,
-		Teammates: teammates,
-	}, nil
+	return data, nil
+}
+
+func (r *frontendRouter) getUserTeamInfo(ctx *gin.Context, user *entities.User) (team *entities.Team, teammates []entities.User, err error) {
+	if user.Team == primitive.NilObjectID {
```

Comment:
Shouldn't we return some sort of error here instead of `nil`?

Reply:
user.Team == primitive.NilObjectID means that the user is not in a team, I don't think we should treat that as an error.

Reply:
Oh right, makes sense!